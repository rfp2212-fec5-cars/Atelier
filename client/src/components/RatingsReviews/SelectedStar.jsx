import React from 'react';

const SelectedStar = ({ stars, handleSortStar }) => {

  return (
    <div className='selected-star' onClick = {()=>handleSortStar(stars)} data-testid = 'selectedstar'>
      <span>{stars} stars
        <span className = 'delete-selected'> âœ•</span>
      </span>
    </div>

  );

};
export default SelectedStar;