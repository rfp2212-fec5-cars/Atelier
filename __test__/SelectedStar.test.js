import TestRenderer from 'react-test-renderer';
import SelectedStar from '../client/src/components/RatingsReviews/SelectedStar.jsx';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { render, screen, fireEvent } from '@testing-library/react';



describe('ProductCharacteristic testing', () => {

  it('should have two span to show selected stars', () => {
    const testRenderer = TestRenderer.create(
      <SelectedStar stars={'5'}></SelectedStar>
    );
    const testInstance = testRenderer.root;
    expect(testInstance.findAll(node => node.type === 'span')).toHaveLength(2);
  });

  test('should invoke handleUserClick once user click', async () => {
    const handleSortStar = jest.fn();
    const utils = render( <SelectedStar stars={'5'} handleSortStar={handleSortStar}></SelectedStar>)
    const sort = utils.getByTestId('selectedstar')
    expect(sort).toBeTruthy();


    fireEvent.click(sort)
    expect(handleSortStar).toHaveBeenCalledTimes(1);

  });

});


