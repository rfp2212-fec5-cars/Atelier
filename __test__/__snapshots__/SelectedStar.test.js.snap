// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCharacteristic testing should exists 1`] = `
<div
  className="selected-star"
  onClick={[Function]}
>
  <span>
    5
     stars
    <span
      className="delete-selected"
    >
       âœ•
    </span>
  </span>
</div>
`;
